### Dictionary

In [12]:
d={1:"alekhya",2:"gireesha"}

In [13]:
d

{1: 'alekhya', 2: 'gireesha'}

In [14]:
del(d)

In [15]:
d

NameError: name 'd' is not defined

In [18]:
t={1:"tejaswini",2:"pallamreddy"}
t

{1: 'tejaswini', 2: 'pallamreddy'}

In [19]:
tc=t.copy()

In [20]:
tc

{1: 'tejaswini', 2: 'pallamreddy'}

In [22]:
tc.get(2)

'pallamreddy'

In [23]:
t.items()

dict_items([(1, 'tejaswini'), (2, 'pallamreddy')])

In [24]:
t.keys()

dict_keys([1, 2])

In [25]:
t.values()

dict_values(['tejaswini', 'pallamreddy'])

In [26]:
t.pop(1)

'tejaswini'

In [27]:
t

{2: 'pallamreddy'}

In [28]:
t.popitem()

(2, 'pallamreddy')

In [29]:
t

{}

In [30]:
t={1:"tejaswini",2:"pallamreddy"}

In [31]:
t.setdefault(3)

In [32]:
t

{1: 'tejaswini', 2: 'pallamreddy', 3: None}

In [33]:
h={'branch':{"ece","eee"},'dis':{"chittor","srikakulam"}}
h

{'branch': {'ece', 'eee'}, 'dis': {'chittor', 'srikakulam'}}

In [35]:
h['branch']['ece']

TypeError: 'set' object is not subscriptable

In [36]:
h["branch"]

{'ece', 'eee'}

In [42]:
h={'branch':{"ece":["cse","eee"]},'dis':{"chittor","srikakulam"}}

In [43]:
h["branch"]["ece"][1]

'eee'

In [61]:
#contact application
contact={}
def createcontact(name,number):
    contact[name]=number
createcontact(input(),input())
print(contact)

alekhya
9876543210
{'alekhya': '9876543210'}


In [62]:
def searchnumber(name,number):
    if name in contact:
        print("contact already exist")
    else:
        contact[name]=number
        print(contact)
searchnumber(input(),input())

alekhya
9876543210
contact already exist


##### task2
* In Contact Application, update already exists contact

   *Input Format:

       Alekhya,76543212445
   *Output Format

       Contact Updated Sucessfully

In [68]:
def updatenumber(name,number):
    if name in contact:
        contact[name]=number
        #print(contact)
        print("updated sucessfully")
    else:
        contact[name]=number
        print("not updated")
updatenumber(input(),input())

alekhya
9849827594
updated sucessfully


## tasks

* Frequency Count: need to print characters with frequency from given sentence (string) in dictionary format.
       *Input Format:

             Andhra Pradesh
     *Output Format:

         {'A': 1,'n': 1,'d': 2,'h': 2,'r': 2,'a': 2,' ': 1,'p': 1,'e': 1,'s': 1}

In [80]:
st=input().lower()
d={}
for el in st:
    if el==" ":
        continue
    elif el not in d:
        d[el]=1
    else:
        d[el]+=1
print(d)

Andhra Pradesh
{'a': 3, 'n': 1, 'd': 2, 'h': 2, 'r': 2, 'p': 1, 'e': 1, 's': 1}


In [83]:
# or
s=input()
sd={}
for char in s:
    if char==" ":
        continue
    else:
         sd[char]=sd.get(char,0)+1
sd

Andhra Pradesh


{'A': 1, 'n': 1, 'd': 2, 'h': 2, 'r': 2, 'a': 2, 'P': 1, 'e': 1, 's': 1}

* Create a nested Dictionary and Access values through the key.

   *Input Format:
            "Alekhya":{"alekhya@gmail.com":9876543210}

   *Output Format:
             9876543210

In [90]:
di={input():{input():input()}}
di[input()][input()]               

alekhya
alekhya@gmail.com
34567
alekhya
alekhya@gmail.com


'34567'